internal/api/create_warehouse_v1alpha1.go

```diff
@@ -0,0 +1,57 @@
+package api
+
+import (
+	"context"
+
+	"connectrpc.com/connect"
+	"github.com/pkg/errors"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	typesv1alpha1 "github.com/akuity/kargo/internal/api/types/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func (s *server) CreateWarehouse(
+	ctx context.Context,
+	req *connect.Request[svcv1alpha1.CreateWarehouseRequest],
+) (*connect.Response[svcv1alpha1.CreateWarehouseResponse], error) {
+	var warehouse kargoapi.Warehouse
+	switch {
+	case req.Msg.GetYaml() != "":
+		if err := yaml.Unmarshal([]byte(req.Msg.GetYaml()), &warehouse); err != nil {
```

Comment:
Minor nitpick, do you think we should also validateProjectAndWarehouseName here to keep it consistent with the other case? The yaml could be missing the name and project too right?